### PR TITLE
Add helm annotations back to PriorityClass

### PR DESCRIPTION
**Pull request by erkanerol**
After discussing the upgrade issue, we decided to add annotations back including `pre-upgrade` annotation.

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: helm/chart-operator/templates/priorityclass.yaml [hunk 1]
```diff
@@ -16,6 +16,11 @@ metadata:
   name: giantswarm-critical
   labels:
     {{- include "chart-operator.labels" . | nindent 4 }}
+  annotations:
+    helm.sh/hook: pre-install
+    helm.sh/hook: pre-upgrade
```
[the remark this answers is not in the document]
Comment (erkanerol) [reply helm/chart-operator/templates/priorityclass.yaml]: Sorry for this stupid mistake. I am testing this version now.

**Comment by erkanerol** [comment]
```
6       	Tue Aug  9 09:33:30 2022	failed    	chart-operator-2.27.0-3ee8bd62c84f785d492878bbe5dfc1fe6c651b07	2.27.1-dev 	Upgrade "chart-operator" failed: pre-upgrade hooks failed: warning: Hook pre-upgrade chart-operator/templates/priorityclass.yaml failed: priorityclasses.scheduling.k8s.io "giantswarm-critical" already exists
```

@ljakimczuk @glitchcrab This doesn't work when this PriorityClass exists.

```
apiVersion: scheduling.k8s.io/v1
description: This priority class is used by giantswarm kubernetes components.
kind: PriorityClass
metadata:
  annotations:
    helm.sh/hook: pre-install
    helm.sh/hook-delete-policy: hook-failed
    helm.sh/resource-policy: keep
  creationTimestamp: "2022-08-04T13:36:58Z"
  generation: 1
  labels:
    app: chart-operator
    app.giantswarm.io/branch: HEAD
    app.giantswarm.io/commit: d3ad0e50c508115b8a8106dc8d508472d275f155
    app.kubernetes.io/instance: chart-operator
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: chart-operator
    app.kubernetes.io/version: 2.27.0
    helm.sh/chart: chart-operator-2.27.0
  name: giantswarm-critical
  resourceVersion: "9885"
  uid: 1ab9c331-2d3a-4920-a61c-e0f9cadb678c
preemptionPolicy: PreemptLowerPriority
value: 1000000000
```

**Comment by erkanerol** [comment]
Tested scenarios:

- [ ] Installation
  - [ ] Priorityclass doesn't exist
  - [ ] Priorityclass exist without annotations + labels
  - [ ] Priorityclass exist without annotations + labels

- [ ] Upgrade
  - [ ] Priorityclass doesn't exist
  - [ ] Priorityclass exist without annotations + labels
  - [ ] Priorityclass exist with annotations + labels

**Review by piontec — APPROVED** [review]
LGTM!

**Review by ljakimczuk — APPROVED** [review]

**Pull request merged**